### PR TITLE
[SH4RE-100] Edit Account

### PR DESCRIPTION
**Pull request by WasifButt**
- Added edit account view to account page 
- this view lets you edit name, pfp, and password
- added profile picture to backend
- user can add profile picture during create account

https://user-images.githubusercontent.com/19561395/226761373-06ffbc30-431b-4b84-a48e-d53e40c36ece.mov

### CODE REVIEW

File: SH4RE/SH4RE/Account/EditAccountView.swift [hunk 1]
```diff
@@ -6,18 +6,190 @@
...
+                Button(action: {
+                    update()
+                }) {
+                    Text("Update")
+                        .fontWeight(.semibold)
+                        .frame(width: screenSize.width * 0.8, height: 20)
+                        .padding()
+                        .foregroundColor(.white)
+                        .background(Color.primaryDark)
+                        .cornerRadius(40)
```
Comment (hannahbrooks) [inline SH4RE/SH4RE/Account/EditAccountView.swift]: no using button style?

File: SH4RE/SH4RE/Account/EditAccountView.swift [hunk 2]
```diff
@@ -6,18 +6,190 @@
...
+                })
+                {
+                    Text("Cancel")
+                        .fontWeight(.semibold)
+                        .frame(width: screenSize.width * 0.8, height: 10)
+                        .padding()
+                        .foregroundColor(.primaryDark)
+                        .background(.white)
+                        .cornerRadius(40)
+                        .overlay(RoundedRectangle(cornerRadius: 40) .stroke(Color.primaryDark, lineWidth: 2))
```
Comment (hannahbrooks) [inline SH4RE/SH4RE/Account/EditAccountView.swift]: same here w button style

**Review by hannahbrooks — DISMISSED** [review]

**Comment by americobarros** [comment]
true, maybe use the button styling. Otherwise looks great!

File: SH4RE/SH4RE/Login/CreateAccountView.swift [hunk 3]
```diff
@@ -29,123 +33,134 @@ struct CreateAccountView: View {
...
+                                .font(.system(size: 18))
+                                .frame(maxWidth: screenSize.width * 0.8, alignment: .leading)
+                            TextField("Your Name", text: $name)
+                                .disableAutocorrection(true)
+                                .frame(width: screenSize.width * 0.8)
+                                .textFieldStyle(textInputStyle())
+                                .padding(.bottom)
+                            Text("Email Address")
+                                .font(.system(size: 18))
+                                .frame(maxWidth: screenSize.width * 0.8, alignment: .leading)
```
Comment (americobarros) [inline SH4RE/SH4RE/Login/CreateAccountView.swift:71]: if you align to . leading, you shouldn't need to put a frame eon all the text. In order to center the photo after making everything leading, you can put the profile photo in a HStack with spacers on both sides, should remove a ton of these frames and manual sizing. Also, something that is really helpful as well is the .fixedSize modifier. Might help in the future

File: SH4RE/SH4RE/Account/AccountView.swift [hunk 4]
```diff
@@ -166,6 +167,14 @@ struct AccountView: View {
                     .onAppear() {
                         getCurrentUser() {(result) in
                             name = result.name
+                            let storageRef = Storage.storage().reference(withPath: result.pfpPath)
+                            storageRef.getData(maxSize: 1 * 1024 * 1024) { [self] data, error in
```
Comment (americobarros) [inline SH4RE/SH4RE/Account/AccountView.swift:170]: this doesnt break the preview right? I know I had to add the "as Int64" for viewlistingview, so just double checking!

**Pull request merged**